### PR TITLE
book 0.7

### PR DESCRIPTION
**Pull request by LeonMatthesKDAB**
This mostly means a thorough review of what we already have.

### CODE REVIEW

File: book/src/bridge/extern_rustqt.md [hunk 1]
```diff
@@ -102,27 +108,22 @@ The `#[qproperty(TYPE, NAME, ...)]` attribute can be specified on a [`#[qobject]
...
-- getter: `get<Property>`
-- changed: `<Property>Changed`
+CXX-Qt will then generate these functions:
 
-On the Rust side:
+|                | C++                | Rust                 |
+|----------------|--------------------|----------------------|
+| setter         | `set<Property>`    | `set_<Property>`     |
+| getter         | `get<Property>`    | `<Property>`         |
+| changed signal | `<Property>Changed`| `<Property>_changed` |
```
Comment (ahayzen-kdab) [inline book/src/bridge/extern_rustqt.md:124]: do we need to now how in C++ for set/get the first letter becomes upper case?

eg `set<MyProperty>` and then `set_<myProperty>` in Rust etc

File: book/src/bridge/extern_rustqt.md [hunk 2]
```diff
@@ -131,24 +132,32 @@ Where `<Property>` is the name of the property.
...
+> **📝 Note**: the key for the flags use all capitals like in the Qt version of qproperty
 
-It is also possible to use any combination of custom functions or omit them entirely, but if flags are specified, read must be included as all properties need to be able to be read.
+It is possible to use any combination of flags or omit some of them entirely, but if any flags are specified, the `READ` flag must be included.
 
-Using the read flag will cause CXX-Qt to generate a getter function with an automatic name based off the property. e.g. `#[qproperty(i32, num, READ)]` will have a getter function generated called get_num in Rust, and getNum in C++.
+If a custom function is specified for a flag, the function must be declared in the bridge and a corresponding implementation must exist.
 
-If a custom function is specified, an implementation both in qobject::MyObject and and export in the bridge is expected.
+Any flag may be passed without specifying a function (e.g. `READ` instead of `READ=...`).
```
Comment (ahayzen-kdab) [inline book/src/bridge/extern_rustqt.md:148]: Any is not correct? Eg `RESET` must have a function specified we have no auto generation option?

And note others like CONSTANT/REQUIRED/FINAL do not have a function

File: book/src/getting-started/5-cmake-integration.md [hunk 3]
```diff
@@ -77,17 +77,12 @@ In the end, your `Cargo.toml` should look similar to this.
 
 [dependencies]
 cxx = "1.0.95"
-cxx-qt = "0.6"
-cxx-qt-lib = "0.6"
+cxx-qt = "0.7"
+cxx-qt-lib = { version="0.7", features = ["qt_full"] }
```
Comment (ahayzen-kdab) [inline book/src/getting-started/5-cmake-integration.md:81]: normally i update all the version at the same time when doing a release. 

And there are two options around that, you either update the version numbers as you tag and do the release, or you pre-emptively update them at the start of the development cycle just after a release. So far we have been updating them when we do a release

File: book/src/bridge/extern_rustqt.md (hunk 1 above)
Comment (LeonMatthesKDAB) [reply book/src/bridge/extern_rustqt.md:124]: Added a footnote and used `<property>` and `<Property>` respectively.

File: book/src/getting-started/3-qml-gui.md [hunk 4]
```diff
@@ -18,7 +18,7 @@ So let's add a `main.qml` file in a `qml` folder:
 ```
 
 If you're not familiar with QML, take a look at the [Qt QML intro](https://doc.qt.io/qt-6/qmlapplications.html).
-We of course also recommend our [QML Intro Training](https://www.kdab.com/software-services/on-site-training/qt-onsite/programming-qtqml-onsite-training/).
+We of course also recommend our [QML Intro Training](https://training.kdab.com/portfolio/introduction-to-qtqml/).
```
[the remark this answers is not in the document]
Comment (LeonMatthesKDAB) [reply book/src/getting-started/3-qml-gui.md:21]: Why not I guess :man_shrugging:

**Review by ahayzen-kdab — APPROVED** [review]
LGTM :-)

**Pull request merged**